src/boot/efifirmware.c

```diff
@@ -0,0 +1,60 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "efifirmware.h"
+#include "util.h"
+
+static const char* efifw_get_fwid(const void *efifwblob) {
+        if ((uintptr_t) efifwblob % alignof(EfiFwHeader) != 0)
+                return NULL;
+
+        const EfiFwHeader *fw_header = ASSERT_PTR(efifwblob);
+
+        if (be32toh(fw_header->magic) != UINT32_C(FWHEADERMAGIC))
+                return NULL;
+
+        uint32_t fw_size = be32toh(fw_header->total_size);
+        uint32_t fw_metadata_off = be32toh(fw_header->fw_metadata_offset);
+        uint32_t fw_metadata_len = be32toh(fw_header->fw_metadata_len);
+        uint32_t fwid_off = be32toh(fw_header->fwid_offset);
+        uint32_t fwid_len = be32toh(fw_header->fwid_len);
+        uint32_t payload_off = be32toh(fw_header->payload_offset);
+        uint32_t payload_len = be32toh(fw_header->payload_len);
```

Comment:
big endian???

Reply:
I took them from `devicetree_get_compatible()` . but I will check and verify.

Reply:
devicetree is a separate spec, they used BE because they come from a BE world. but this stuff here is defined by us, by you, in an efi world, where everything is LE.